look-and-feel/react/src/ModalClient/type.ts

```diff
@@ -0,0 +1,43 @@
+import { ReactNode, RefObject } from "react";
+
+export type ButtonAction = {
+  text: string;
+  disabled?: boolean;
+  callback: () => void;
+};
+
+export type TActionButton = {
+  primary?: ButtonAction;
+  secondary?: ButtonAction;
+  tertiary?: ButtonAction;
+};
+
+export type TModalHook = {
+  setIsOpen: (value: boolean) => void;
+  onClose?: () => void;
+  onClickOutside?: () => void;
+};
+
+export type ModalProps = TModalHook & {
+  isOpen: boolean;
```

Comment:
On essaie de garder le nom des props des composants HTML natifs, et dans une modale c'est `open`
```suggestion
  open: boolean;
```